blocks/serialization.py

```diff
@@ -180,19 +212,150 @@ def secure_dump(object_, path, dump_function=dump, **kwargs):
+        The name of the object you are dumping. It will be used as a file
+        name in the archive. '_pkl' and '_paramters' are reserved names
+        and can't be used.
+    parameters : list, optional
+        Shared variables whose internal numpy arrays should be saved
+        separately in the `_parameters` field of the tar file. Must be a
+        subset of the parameters already in the archive.
+    use_cpickle : bool
+        Use cPickle instead of pickle. Setting it to true will disable the
+        warning message if you try to pickle objects from the main module!
+        Be sure that you don't have the warning before turning this flag
+        on. Default: False.
+    protocol : int, optional
+        The pickling protocol to use. Unlike Python's built-in pickle, the
+        default is set to `2` instead of 0 for Python 2. The Python 3
+        default (level 3) is maintained.
+    \*\*kwargs
+        Keyword arguments to be passed to `pickle.Pickler`.
+
+    """
+    if name in ['_pkl', '_parameters']:
+        raise ValueError("_pkl and _parameters are reserved names and can't" \
+                         " be used as name for your object.")
+
+    external_parameters = {}
+    if parameters is not None:
+        renamer = _Renamer()
+        named_parameters = {renamer(p): p for p in parameters}
+        for n, p in named_parameters.items():
+            array_ = p.container.storage[0]
+            external_parameters[id(array_)] = _mangle_parameter_name(
+                type(array_), n)
+
+        # Check that the parameters are the same that the ones in the archive.
+        with closing(tarfile.TarFile(fileobj=file_, mode='r')) as tar_file:
+            if '_parameters' not in tar_file.getnames():
+                raise ValueError("There is no parameters in the archive, so" \
+                                 " you can't use the argument parameters.")
+            else:
+                parameters = numpy.load(
+                    tar_file.extractfile(tar_file.getmember('_parameters')))
+                s1 = set(parameters.keys())
+                s2 = set(external_parameters.keys())
+                if s1.issuperset(s2):
+                    raise ValueError('The set of parameters is different' \
+                                     ' from the one in the archive.')
+
+    # TODO: How to avoid this hack?
+    file_.close()
+    file_ = open(file_.name, 'r+')
+
+    if use_cpickle:
+        pickler = cPickle.Pickler
+    else:
+        pickler = _PicklerWithWarning
+    with closing(tarfile.TarFile(fileobj=file_, mode='a')) as tar_file:
+        def _save_object(f):
+            p = pickler(f, protocol=protocol, **kwargs)
+            p.persistent_id = _PersistentID(external_parameters)
+            p.dump(object_)
+        _taradd(_save_object, tar_file, name)
+
+
+def continue_training(file_):
     """Continues training using checkpoint.
 
     Parameters
     ----------
-    path : str
+    file_ : str
```

Comment:
The same question here, I think `path` is a more informative name when a string is expected.